client/state/selectors/is-eligible-for-upwork-support.ts

```diff
@@ -12,22 +12,22 @@ import { getCurrentUserLocale } from 'state/current-user/selectors';
 import getSitesItems from 'state/selectors/get-sites-items';
 import { isBusinessPlan, isEcommercePlan } from 'lib/plans';
 
-const SPANISH_LANG_SLUGS = [ 'es', 'es-cl', 'es-mx' ];
+const UPWORK_LOCALES = [ 'es', 'es-cl', 'es-mx', 'fr', 'fr-ca', 'fr-be', 'fr-ch' ];
 
 /**
  * @param {Object} state Global state tree
  * @return {Boolean} Whether or not this customer should receive Upwork support
  */
-export default function isEligibleForUpworkSupport( state ) {
-	// Upwork is currently available for Spanish-speaking users only
-	if ( ! includes( SPANISH_LANG_SLUGS, getCurrentUserLocale( state ) ) ) {
+export default function isEligibleForUpworkSupport( state ): boolean {
```

Comment:
The only explicit typing in this change.

I'm not super clear on how to test the typing in the context of the way it's being introduced into Calypso from the command line. Is it...?

`npx tsc client/state/selectors/is-eligible-for-upwork-support.ts`

Reply:
that might give you a mountain of errors.

personally I've _only_ been using my IDE and its toolips, inline errors, autocomplete, and other aids to interact with typed code.

if we go back to `tsc` it's going to want to point out everything, and if your module has very few type errors that might work for you, but in the code I've worked in they are all over the place and so I have to kind of prioritize what I pay attention to.

does that help?

Reply:
Gotcha. Welp, I'm not in the Calypso codebase enough to throw a super strong opinion. FWIW, IMO, It'd be nice if "flat file editor" folk had a CLI option.

Reply:
Not sure what you mean by "flat file editor" - feel free to elaborate. The CLI option is `tsc` - I'm just saying you'll get lots of noise.

For text-mode IDEs you can refer to this non-exhaustive list:
 - `vim` https://github.com/Quramy/tsuquyomi
 - `emacs` https://github.com/ananthakumaran/tide/

Here's the list of [supported IDEs](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Editor-Support) from their website.

If by "flat file editor" you mean `ed` and `grep` then sorry man, I can't help you.

Reply:
I only write code with `patch`.